### DISCUSSION TITLE
Hasura integrated with Vercel's CI/CD pipline

### DISCUSSION THREAD

**Discussion by nbouvrette** [hasura/graphql-engine#8718] [Question] [answered]
We've been using Vercel for a while and one of its nice features is that it creates a "preview" instance of your app every time you commit code on Git.

One of the challenges is that each "preview" instance is pointing to the prod Hasura. In fact, what I would be looking for is a way to spin "Hasura previews" and align them with the "Vercel previews". Meaning if we make database changes, they would automatically be reflected in the preview and later promoted to production using migration.

I found this Vercel integration document [here](https://hasura.io/docs/latest/projects/vercel-integration/) but it doesn't look like it's doing this at all - is there such a solution available out there or would we need to develop it from scratch?

**Comment by meetzaveri** [comment, marked as answer]
Hello @nbouvrette,

**Regarding your query:**
Thanks for asking. Reading your question, I can roughly understand - that whenever you want to commit code , lets say you create a PR. Upon creation of PR, you want to check if everything is working as expected with the new code by creating a preview app from metadata and migrations.

**Intro to hasura cloud preview apps:**
If this your use case, then we do have feature called hasura cloud preview apps. Please check the docs here for more clarity - https://hasura.io/docs/latest/deployment/preview-apps/. You can create a simple action (it will be `.yaml` file) and you deploy it with your hasura data (i.e. metadata and migrations). You can check how to structure your action here - https://github.com/marketplace/actions/hasura-cloud-preview-apps . The preview apps internal code is also open sourced here - https://github.com/hasura/hasura-cloud-preview-apps .

So whenever you'll create a PR, there will be jobs running in background using your action file. If that job run is successful, that means there will be a new hasura cloud project created with that metadata and migrations every time you open a PR against your codebase. 

This hasura cloud project will be created under your cloud hasura dashboard account. For creating account, you'll need to sign up for hasura cloud. You can either create free tier or paid tier account.

I hope this was helpful. Do let us know if this doesn't match your use case. Cheers!

**Reply by nbouvrette** [reply]
Yes, in the case of Vercel, they build previews on every commit (or you can filter the name of the branches) but I think your solution should be able to help with this.

Thanks you
